egret/model_library/unit_commitment/objective.py

```diff
@@ -117,6 +118,8 @@ def compute_total_production_cost_rule(m, t):
         supp = True
     if hasattr(model, 'flexible_ramping'):
         flex = True
+    if hasattr(model, 'security_constraints'):
```

Comment:
I am somewhat opposed to using `hasattr` to infer things like this rather than being explicit.

Reply:
So, I've made things slightly more explicit by adding this attributes to the model no matter what. But at some level, this objective function builder either has to infer things about the state of the model, or take in potentially several flags.